DESIGN.md

```diff
@@ -0,0 +1,60 @@
+# Bulk Data Extract (BDE) Processor design
+
+## User requirements
+
+-  The _user interface_ is a read-only PostgreSQL database in AWS RDS, with
+   Azure authentication
+-  _Accessible_ to all
+   [current BDE users](https://github.com/linz/ds-infra-team/wiki/BDE-users)
+-  Process BDE files
+   _[every day](https://toitutewhenua.atlassian.net/wiki/spaces/GEOD/pages/86418416/BDE+Processor+Product)
+   except Sunday_
+-  Optional _post-processing_ pulled in from separate internal repository
+-  All changes for a single day are imported within a _single transaction_
+-  _Immediate notification_ of completed import (like the legacy BDE Processor
+   SNS topic)
+
+## Technical requirements
+
+-  Thorough testing of every part of the process before merging any pull
+   requests
+-  Separate, managed accounts for infrastructure testing and production
+-  Trivial deployment
+
+## Open questions
+
+-  Can we give read access to the BDE data to the entire organisation? The
+   public?
+
+## Non-requirements
+
+-  Don't keep BDE files around. They should ideally be kept only in memory or at
+   most in temporary storage which is cleaned up after each run. If we
+   absolutely need to keep them around that should be done as a separate system.
+   If it's convenient, the BDE Processor could read from the data in that system
+   rather than doing its own data copying.
+-  No table versioning. It creates a _lot_ of extra complexity, and we don't
+   have any PostgreSQL experts at hand to maintain the extensions.
+
+## Wanted features
+
+-  Pull rather than push data from Landonline. This way we can keep the entire
+   system together, rather than having a costly and awkward split between
+   on-premise and cloud architecture.
+-  Continuous deployment. Any time a change is merged to the main branch some
+   smoke tests are performed and the result is deployed to production.
+-  React to change rather than run on a timer, where possible. Ideally the BDE
+   Processor should know about new files as soon as the last file has been
+   closed for writing, and should immediately start processing.
+-  Deal with upstream schema changes without code changes.
```

Comment:
I like this one, although in reality this only happens once every couple of years.
Also need to consider downstream users like the LINZ Data Service and the end users of the data. So whiel we might not need ot change code (a good thing) we would probably have to do something with the customers and open data team.